controllers/etcd_controller.go

```diff
@@ -179,11 +180,29 @@ func (r *EtcdReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	if !reflect.DeepEqual(etcd.Spec, etcdCopy.Spec) {
 		etcdCopy.Spec = etcd.Spec
 	}
-	logger.Infof("Reconciling etcd: %s/%s", etcd.GetNamespace(), etcd.GetName())
+
+	etcdCopy.Status.ObservedGeneration = &etcdCopy.Generation
+	if _, ok := etcd.Annotations[common.GardenerOperation]; ok {
+		delete(etcd.Annotations, common.GardenerOperation)
+		if err := r.Update(context.TODO(), etcd); err != nil {
```

Comment:
Move the update call out of the `if operation annotation is set` condition. The observed generation should also we updated.

Reply:
Update does not update the `status` sub-resource. If update failed, it would still reconcile as part of retries. Do I need to explicitly update it here. I am updating the status eventually.

Reply:
OK, I see. Why not setting the observedGeneration then in your updateStatus functions like you set all the other fields there?